Automattic/woocommerce-payments
# refactor: enable eslint hook rules

Fixes #

#### Changes proposed in this Pull Request









Discussed on Slack: p1621001462390700-slack-CGGCLBN58

Enabling the following eslint rules:
- `react-hooks/exhaustive-deps` : ensures that all the necessary dependencies are passed to hooks
- `react-hooks/rules-of-hooks` : ensures that hooks aren't called conditionally or within other hooks

Refactored the code (where I knew I could do it with confidence) to fix the rules.

#### Testing instructions





Most of the changes affect the "grouped settings" work, behind a feature flag:
- Ensure you have the "grouped settings" flag enabled in WCPay Dev Tools
- Visit http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout

-------------------
